sphinx/builders/gettext.py

```diff
@@ -91,13 +91,15 @@ def __init__(self, source: str, line: int) -> None:
 
 class GettextRenderer(SphinxRenderer):
     def __init__(
-        self, template_path: list[str | os.PathLike[str]] | None = None,
+        self, template_path: Sequence[str | os.PathLike[str]] | None = None,
             outdir: str | os.PathLike[str] | None = None,
     ) -> None:
         self.outdir = outdir
-        if template_path is None:
-            template_path = [path.join(package_dir, 'templates', 'gettext')]
-        super().__init__(template_path)
+        default_template_path: str = path.join(package_dir, 'templates', 'gettext')
+        template_dirs: list[str | os.PathLike[str]] = [default_template_path]
+        if template_path is not None:
+            template_dirs = [*list(template_path), default_template_path]
```

Comment:
```python
            template_dirs = [*list(template_path), default_template_path]
```

I'm wondering about the implications of including `default_template_path` implicitly here, if `template_path` is already set.

Reply:
Benefits / disadvantages of the implicit default:

 * :heavy_plus_sign: Projects that build into multiple formats including `gettext`, and that have already configured [`templates_path`](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-templates_path), will continue to build successfully.
 * :heavy_minus_sign: This behaviour is inconsistent with our other builders; for those, when [`templates_path`](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-templates_path) is set, it's the authoritative list of locations that template content is read from.

Reply:
The `default_template_path` here is just the directory that was already being used to look for this particular template, and is only being appended to the directory search path for rendering gettext *.pot files.  As demonstrated by the unit test, when `template_path` is set it takes precedence over the default location for this particular template.